pytest_stf/plugin.py

```diff
@@ -87,47 +99,77 @@ def fixture_allocated_phone(pytestconfig):
     requirements = pytestconfig.getoption('phone_requirements')
-    timeout = pytestconfig.getoption('stf_allocation_timeout')
     assert requirements, 'phone_requirements required'
-    requirements = parse_requirements(requirements)
+    assert 'platform' in requirements, 'platform required'
+
+    if hasattr(pytestconfig, '_openstf'):
+        stf = pytestconfig._openstf  # pylint: disable=protected-access
+        stf  # type=StfClient
+        timeout = pytestconfig.getoption('stf_allocation_timeout')
+        requirements = parse_requirements(requirements)
 
-    with stf.allocation_context(requirements, timeout_seconds=timeout) as device:
-        yield device
+        with stf.allocation_context(requirements, timeout_seconds=timeout) as device:
+            yield device
+    else:
+        allocation_timeout = pytestconfig.getoption('allocation_timeout') or 0
+        with lockable.auto_lock(requirements, allocation_timeout) as device:
+            yield device
 
 
 @pytest.fixture(name='phone_with_adb', scope="session")
-def fixture_phone_with_adb(allocated_phone):
+def fixture_phone_with_adb(pytestconfig, allocated_phone):
     """
-    Allocate required phone and create ADB tunnel to phone via STF.
+    1) Allocate a required phone via STF or lockable
+    2) Create an ADB tunnel to phone if using STF
     Yields tuple (adb: AdbServer, device: dict)
     """
-    with AdbServer(allocated_phone['remote_adb_url']) as adb:
-        yield adb, allocated_phone
+    if hasattr(pytestconfig, '_openstf'):
+        with AdbServer(allocated_phone['remote_adb_url']) as adb:
+            yield adb, allocated_phone
+    else:
+        yield None, allocated_phone
 
 
 @pytest.fixture(name='appium_args', scope='session')
-def fixture_appium_args():
+def fixture_appium_args(pytestconfig):
     # overridable list of appium server args
-    return []
+    args = []
+    logs_file = pytestconfig.getoption('appium_logs')
+    if logs_file:
+        args.extend(['--log', logs_file])
+    return args
+
 
 @pytest.fixture(name="appium_server", scope="session")
-def fixture_appium_server(phone_with_adb, appium_args):
+def fixture_appium_server(pytestconfig, phone_with_adb, appium_args):
     """
-    Allocate required phone, create ADB tunnel to phone via STF and start appium server for tests.
-    Yields tuple (appium: Appium, adb: AdbServer, device: dict )
+    1) Allocate a required phone via STF or lockable
+    2) Create an ADB tunnel to phone if using STF
+    3) Start Appium server or use remote one if requested
+    Yields tuple (appium: AppiumServer|RemoteAppiumServer, adb: AdbServer, device: dict )
     """
     adb, phone = phone_with_adb
-    with AppiumServer(appium_args) as appium:
-        yield appium, adb, phone
+    if adb:
+        with AppiumServer(appium_args) as appium:
+            yield appium, adb, phone
+    else:
+        appium_api_path = phone.get(
+            'appium_server') or pytestconfig.getoption('appium_server')
```

Comment:
phone:appium_server was not mentioned in pr description/help, right?

Reply:
added mention of it to README